### ISSUE TITLE
Do skip and todo with exceptions

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2162]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2162|
|Type|task|
|Created|2006-10-10 07:38:52Z|

```
result.addError(test, SkipTest("reason"))
result.addError(test, TodoTest("reason", trappedFailure))
result.addError(test, DoneTest(trappedFailure))
```

Vaguey McVague -- I know what I mean.

<details><summary>Searchable metadata</summary>

```
trac-id__2162 2162
type__task task
reporter__jml jml
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__invalid invalid
component__trial trial
keywords__ 
time__1160465932000000 1160465932000000
changetime__1366581263000000 1366581263000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

This looks like it covers about half of [#2411](https://github.com/twisted/twisted/issues/2411).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

How does this relate to [#4596](https://github.com/twisted/twisted/issues/4596)?

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_4"></a>|@Julian commented|
|-|-|

The skip portion of this appears to be done, since unittest grew an exception for `SkipTest` in 2.6.

The todo portion of this is outstanding but might be addressed alongside [#4811](https://github.com/twisted/twisted/issues/4811).

Done would seem odd to do with an exception, though it probably should be deprecated and replaced with `stopTestRun` anyhow.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

"Do" is an insufficiently specific verb for an entire ticket to rest on.

It's not clear what this means.  Please file a new ticket if there's some real feature beyond [#4811](https://github.com/twisted/twisted/issues/4811) that's desired.

Selecting the "invalid" resolution because of the lack of clarity.
